genmap.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+worldfile=$(python map_config/get_pkg_path.py $2)
+config=`basename $1`
+worldbody=`xpath -q -e '/sdf/world/*' $worldfile`
+template="""
+<%
+%>
+<sdf version='1.6'>
+  <world name='default'>
+    $worldbody
+    <plugin name='gazebo_occupancy_map' filename='libgazebo_2Dmap_plugin.so'>
+         <map_resolution><%= @map_res %></map_resolution>
+         <map_height>0.3</map_height>
+         <map_size_x><%= @map_size_x %></map_size_x>
+         <map_size_y><%= @map_size_y %></map_size_y>
+         <init_robot_x><%= @robot_init_x %></init_robot_x>
+         <init_robot_y><%= @robot_init_y %></init_robot_y>
+     </plugin>
+ </world>
+</sdf>
+"""
+echo $template | erb -r "./map_config/$config"  | xmllint --format - > "simulation_ws/src/cloudwatch_simulation/worlds/map_plugin.world"
```

Comment:
Same here. It's simpler to just output the change -- leave this script/tool naive and all it does is substitute a template into a world file. The user can redirect. If we want wrappers, we can wrap this underlying tool. 

```
./genmap.sh <config_file> <world_file> > <new_world_file>
```

Reply:
The output was hard coded to make the launch file to start the map_service generic and user-friendly. Giving the end-user the liberty to direct the output, adds one more user-unfriendly step to reference that world file in the map service launch file. However, I agree that the plugin tool for it's user (maybe not the end-user) could be naive and generic and wrapping it would be a better alternative in that case.